src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py

```diff
@@ -4,43 +4,143 @@
+        self.DEFAULT_RELEASE_TRAIN = 'stable'
+        self.DEFAULT_CLUSTER_TYPE = 'managedclusters'
+        self.DEFAULT_HA = 'true'
+
         # constants for configuration settings.
-        self.CLUSTER_TYPE = 'global.clusterType'
+        self.CLUSTER_TYPE_KEY = 'global.clusterType'
+        self.HA_KEY_ENABLED_KEY = 'global.ha.enabled'
+        self.SKIP_EXISTING_DAPR_CHECK_KEY = 'skipExistingDaprCheck'
+        self.EXISTING_DAPR_RELEASE_NAME_KEY = 'existingDaprReleaseName'
+        self.EXISTING_DAPR_RELEASE_NAMESPACE_KEY = 'existingDaprReleaseNamespace'
+
+        # constants for message prompts.
+        self.MSG_IS_DAPR_INSTALLED = "Is Dapr already installed in the cluster?"
+        self.MSG_ENTER_RELEASE_NAME = "Enter the Helm release name for Dapr, "\
+            f"or press Enter to use the default name [{self.DEFAULT_RELEASE_NAME}]: "
+        self.MSG_ENTER_RELEASE_NAMESPACE = "Enter the namespace where Dapr is installed, "\
+            f"or press Enter to use the default namespace [{self.DEFAULT_RELEASE_NAMESPACE}]: "
+        self.MSG_WARN_EXISTING_INSTALLATION = "The extension will be installed on your existing Dapr installation. "\
+            f"Note, if you have updated the default values for global.ha.* or dapr_placement.* in your existing "\
+            "Dapr installation, you must provide them in the configuration settings. Failing to do so will result in"\
+            "an error, since Helm upgrade will try to modify the StatefulSet."\
+            f"Please refer to {self.TSG_LINK} for more information."
+
+        self.RELEASE_INFO_HELP_STRING = "The Helm release name and namespace can be found by running 'helm list -A'."
+
+        # constants for error messages.
+        self.ERR_MSG_INVALID_SCOPE_TPL = "Invalid scope '{}'. This extension can be installed only at 'cluster' scope."\
+            f" Check {self.TSG_LINK} for more information."
+
+    def _get_release_info(self, release_name: str, release_namespace: str, configuration_settings: dict)\
+            -> Tuple[str, str, bool]:
+        '''
+        Check if Dapr is already installed in the cluster and get the release name and namespace.
+        If user has provided the release name and namespace in configuration settings, use those.
+        Otherwise, prompt the user for the release name and namespace.
+        If Dapr is not installed, return the default release name and namespace.
+        '''
+        name, namespace, dapr_exists = release_name, release_namespace, False
+
+        # Set the default release name and namespace if not provided.
+        name = name or self.DEFAULT_RELEASE_NAME
+        namespace = namespace or self.DEFAULT_RELEASE_NAMESPACE
+
+        if configuration_settings.get(self.SKIP_EXISTING_DAPR_CHECK_KEY, 'false') == 'true':
+            logger.info("%s is set to true, skipping existing Dapr check.", self.SKIP_EXISTING_DAPR_CHECK_KEY)
+            return name, namespace, False
```

Comment:
If the user passes in the true to skip-existing, and if there was an existing instance of Dapr, what will be the user experience during failure?  a. Will the user be able to discover this issue from the error messages that will be returned?  b. Will the user be able to recover from this state?  Is it as simple as rerunning with the skip-existing False - how will they discover the steps for recovery?  May be capture this in the TSG link?

Reply:
Yes, the user will see an error message and the link to our TSG, which points them to the right steps.
Yes, they just need to rerun the operation and provide the correct details (name and namespace).
Our TSGs are being updated with this information.